### DISCUSSION TITLE
Reading Excel 'Tables'

### DISCUSSION THREAD

**Discussion by jgranduel** [duckdb/duckdb#8749] [Ideas]
Hi,

I have seen that Duckdb is able to import Excel data, thanks to GDAL project [https://duckdb.org/docs/guides/import/excel_import.html](https://duckdb.org/docs/guides/import/excel_import.html)

As of now, it loads data from whole sheets if I am not wrong.  More and more data are organised in [Excel tables](https://support.microsoft.com/en-gb/office/overview-of-excel-tables-7ab0bb7d-3a9e-4b56-a3c9-6c94334e492c) which are relatively close to what a database would expect. They are tied with tools like PowerQuery. Excel tables are *named*.

Would it be possible to import tables per name as it's possible to load a range in a Sheet at the moment or all the tables of an Excel Workbook into different relations maybe, named after their respective name in Excel by default?

Thanks in advance, 
best regards

**Comment by yoonghm** [comment]
I came across this solution from

https://stackoverflow.com/questions/54241345/pandas-read-a-table-from-excel#comment95307315_54241345

```Python
import pandas as pd
import openpyxl

def read_table(file_name: str, table_name: str) -> pd.DataFrame:
    wb = openpyxl.load_workbook(file_name, read_only= False, data_only = True) # openpyxl does not have table info if read_only is True; data_only means any functions will pull the last saved value instead of the formula
    for sheetname in wb.sheetnames: # pulls as strings
        sheet = wb[sheetname] # get the sheet object instead of string
        if table_name in sheet.tables: # tables are stored within sheets, not within the workbook, although table names are unique in a workbook
            tbl = sheet.tables[table_name] # get table object instead of string
            tbl_range = tbl.ref #something like 'C4:F9'
            break # we've got our table, bail from for-loop
    data = sheet[tbl_range] # returns a tuple that contains rows, where each row is a tuple containing cells
    content = [[cell.value for cell in row] for row in data] # loop through those row/cell tuples
    header = content[0] # first row is column headers
    rest = content[1:] # every row that isn't the first is data
    df = pd.DataFrame(rest, columns = header)
    wb.close()
    return df
```
